types/wordpress__blocks/index.d.ts

```diff
@@ -503,3 +504,20 @@ export type Transform<T extends Record<string, any> = Record<string, any>> =
     | TransformPrefix<T>
     | TransformRaw<T>
     | TransformShortcode<T>;
+
+export type BlockVariationScope = 'block' | 'inserter' | 'transform';
+
+export interface BlockVariation<T extends Record<string, any> = any> {
+    name: string;
+    title: string;
+    description?: string;
+    category?: string;
+    icon?: BlockIcon;
+    isDefault?: boolean;
+    attributes?: T;
+    innerBlocks?: any[];
```

Comment:
we should probably avoid `any` here. even if all we do is alias a type for the block it's an odd statement that `innerBlocks` can be anything.

we have `BlockInstance` from `@wordpress/blocks` we could import, or we could punt on the issue and give a name.

```ts
interface Block {}

…

interface BlockVariation<…> {
    …
    innerBlocks?: Block[];
    …
}
```

Reply:
Went through the docs and code again and pulled it out as InnerBlockConfiguration with proper typing for nested array and passing block attributes configuration and further nested innerBlocks.